azkaban-common/src/main/java/azkaban/scheduler/Schedule.java

```diff
@@ -341,7 +378,7 @@ public void createAndSetScheduleOptions(Object obj) {
   }
 
   public boolean isRecurring() {
-    return period == null ? false : true;
+    return period == null && cronExpression == null ? false : true;
```

Comment:
can simply to `return period != null || cronExpression != null;`

If you use IntelliJ, it automatically suggests these refactors.